### PR TITLE
Update release workflow

### PR DESCRIPTION
**Pull request by andrzejewsky**
### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `data-test-id` are added for new elements
4. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
5. [ ] Your code works with the latest stable version of the core
6. [ ] I added changesets and [read good practices](../.changeset/README.md)

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] app
3. [ ] attribute
4. [ ] category
5. [ ] collection
6. [ ] customer
7. [ ] giftCard
8. [ ] homePage
9. [ ] login
10. [ ] menuNavigation
11. [ ] navigation
12. [ ] orders
13. [ ] pages
14. [ ] payments
15. [ ] permissions
16. [ ] plugins
17. [ ] productType
18. [ ] products
19. [ ] sales
20. [ ] shipping
21. [ ] translations
22. [ ] variants
23. [ ] vouchers

CONTAINERS=2

### CODE REVIEW

File: .github/workflows/prepare-release.yml [hunk 1]
```diff
@@ -33,3 +35,58 @@ jobs:
...
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: ".nvmrc"
+      - name: Install Dependencies
+        run: npm ci
+      - name: Build
+        run: |
+          npm run build
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
```
Comment (NyanKiyoshi) [inline .github/workflows/prepare-release.yml:69]: I suggest upgrading to v4 as v1 uses deprecated features and runtime. But it's a bigger issue as it needs to be ported all the way down to v3.11, up to you if you want to do it in this PR or another PR

File: .github/workflows/prepare-release.yml [hunk 2]
```diff
@@ -33,3 +35,58 @@ jobs:
           publish: "npm run release"
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  release:
+    name: Deploy and release to sandbox
+    needs: changesets
+    if: needs.prepare.outputs.published == 'true'
+    runs-on: ubuntu-latest
```
Comment (NyanKiyoshi) [inline .github/workflows/prepare-release.yml:43]: I recommend freezing the ubuntu version so it doesn't suddenly stop working
```suggestion
    runs-on: ubuntu-22.04
```

File: .github/workflows/prepare-release.yml [hunk 3]
```diff
@@ -9,9 +9,11 @@ on:
 concurrency: ${{ github.workflow }}-${{ github.ref }}
 
 jobs:
-  release:
+  prepare:
     name: Prepare release with Changesets
     runs-on: ubuntu-latest
```
Comment (NyanKiyoshi) [inline .github/workflows/prepare-release.yml:14]:
```suggestion
    runs-on: ubuntu-22.04
```

**Review by NyanKiyoshi — DISMISSED** [review]

File: .github/workflows/prepare-release.yml (hunk 1 above)
Comment (andrzejewsky) [reply .github/workflows/prepare-release.yml:69]: change it here, but in terms of other versions will do separate PRs

**Review by NyanKiyoshi — APPROVED** [review]

**Review by Droniu — APPROVED** [review]
